components/brave_wallet/browser/nft_metadata_fetcher.cc

```diff
@@ -249,14 +249,15 @@ void NftMetadataFetcher::OnGetTokenMetadataPayload(
   }
 
   // Invalid JSON becomes an empty string after sanitization
-  if (api_request_result.body().empty()) {
+  if (api_request_result.value_body().is_none()) {
     std::move(callback).Run(
         "", static_cast<int>(mojom::JsonRpcError::kParsingError),
         l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR));
     return;
   }
 
-  std::move(callback).Run(api_request_result.body(), 0, "");  // 0 is kSuccess
+  std::move(callback).Run(api_request_result.SerializeBodyToString(), 0,
```

Comment:
That should be TakeValue, but needs some refactoring. Could you please drop a TODO here?

Reply:
Done